unlock-js/src/__tests__/walletService.test.js

```diff
@@ -152,110 +144,127 @@ describe('WalletService', () => {
     })
 
     describe('sendTransaction', () => {
-      const to = ''
-      const from = '0x'
+      const to = '0x0987654321098765432109876543210987654321'
+      const from = '0x1234567890123456789012345678901234567890'
       const data = ''
       const value = ''
-      const gas = ''
+      const gas = '0x1234'
       const privateKey = null
       const contract = {
         abi: [],
       }
-      const mockSendTransaction = jest.fn()
-      let mockTransaction
+      const transaction = {
+        status: 'mined',
+        createdAt: new Date().getTime(),
+        hash:
+          '0x83f3e76db42dfd5ebba894e6ff462b3ae30b5f7bfb7a6fec3888e0ed88377f64',
+      }
 
-      beforeEach(() => {
-        mockTransaction = new EventEmitter()
-        mockSendTransaction.mockReturnValue(mockTransaction)
-        walletService.web3.eth.sendTransaction = mockSendTransaction
-      })
+      describe('success', () => {
+        beforeEach(() => {
+          nock.ethGetGasPriceAndYield('0x123')
+          nock.ethSendTransactionAndYield(
+            { to, from, data, value: '0x0', gas },
+            '0x123',
+            transaction.hash
+          )
+          nock.ethGetTransactionReceipt(transaction.hash, {
+            status: 1,
+            transactionHash: transaction.hash,
+            transactionIndex: '0x0d',
+            blockHash:
+              '0xdc7c95899e030f3104871a597866767ec296e948a24e99b12e0b251011d11c36',
+            blockNumber: `0x${(18).toString('16')}`,
+            contractAddress: '0xcfeb869f69431e42cdb54a4f4f105c19c080a601',
+            gasUsed: '0x16e360',
+            cumulativeGasUsed: '0x16e360',
+            logs: [],
+          })
+        })
 
-      it('should handle cases where the transaction is sent via a provider', () => {
-        expect.assertions(1)
+        it('should handle cases where the transaction is sent via a provider', async () => {
+          expect.assertions(0)
```

Comment:
hu, wat? This is probabluy not a good test if there are no assertions ;p

Reply:
I'll read up on nock, see if we can get something to assert against from it